go/vt/vtgate/executor.go

```diff
@@ -57,6 +57,9 @@ import (
 var (
 	errNoKeyspace     = vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "no keyspace in database name specified. Supported database name format (items in <> are optional): keyspace<:shard><@type> or keyspace<[range]><@type>")
 	defaultTabletType topodatapb.TabletType
+
+	queriesProcessed = stats.NewCountersWithSingleLabel("QueriesProcessed", "Queries processed at vtgate by plan type", "Plan")
```

Comment:
Oh you don't have to enumerate all of the tags you're going to be using?

Reply:
No -- you only need to enumerate if you want them to always be present in the exported metrics with 0 value per tag.

With this interface it will lazily populate the map as different plans are being used.

Reply:
There's subtle bug to be aware of: some monitoring systems differentiate an absent value from 0 value. This means that a new value showing up as `1` is not the same as a value changing from `0` to `1`.

The specific use case where this failed: if there is any change in internal errors, send an alert. Not pre-initializing those values resulted in no alerts being sent.

Reply:
Sure -- that makes sense.

IMO that concern doesn't warrant the extra verbosity of enumerating all possible plan types as well as DDL, Other, etc just to preinitialize the counts to 0.